system/include/SSE/emmintrin.h

```diff
@@ -0,0 +1,1621 @@
+_mm_setr_epi16(short w0, short w1, short w2, short w3, short w4, short w5, short w6, short w7)
+{
+  return (__m128i)wasm_i16x8_make(w0, w1, w2, w3, w4, w5, w6, w7);
+}
+
+static __inline__ __m128i __attribute__((__always_inline__, __nodebug__))
+_mm_setr_epi8(char b0, char b1, char b2, char b3, char b4, char b5, char b6, char b7, char b8, char b9, char b10, char b11, char b12, char b13, char b14, char b15)
+{
+  return (__m128i)wasm_i8x16_make(b0, b1, b2, b3, b4, b5, b6, b7, b8, b9, b10, b11, b12, b13, b14, b15);
+}
+
+static __inline__ __m128i __attribute__((__always_inline__, __nodebug__))
+_mm_setzero_si128(void)
+{
+  return wasm_i64x2_const(0, 0);
+}
+
+static __inline__ void __attribute__((__always_inline__, __nodebug__))
+_mm_store_si128(__m128i *__p, __m128i __b)
+{
+  *__p = __b;
+}
+
+static __inline__ void __attribute__((__always_inline__, __nodebug__))
+_mm_storeu_si32(void *__p, __m128i __a)
+{
+  *(unsigned int *)__p = wasm_i32x4_extract_lane((v128_t)__a, 0);
+}
+
+static __inline__ void __attribute__((__always_inline__, __nodebug__))
+_mm_storeu_si128(__m128i *__p, __m128i __b)
+{
+  struct __unaligned {
+    __m128i __v;
+  } __attribute__((__packed__, __may_alias__));
+  ((struct __unaligned *)__p)->__v = __b;
+}
+
+static __inline__ void __attribute__((__always_inline__, __nodebug__))
+_mm_maskmoveu_si128(__m128i __d, __m128i __n, char *__p)
+{
+  // TODO: optimize
+  union {
+    unsigned char x[16];
+    __m128i m;
+  } mask, data;
+  mask.m = __n;
+  data.m = __d;
+  for(int i = 0; i < 16; ++i)
+    if (mask.x[i] & 0x80)
+      __p[i] = data.x[i];
+}
+
+static __inline__ void __attribute__((__always_inline__, __nodebug__))
+_mm_storel_epi64(__m128i *__p, __m128i __a)
+{
+  *(long long *)__p = wasm_i64x2_extract_lane((v128_t)__a, 0);
+}
+
+static __inline__ void __attribute__((__always_inline__, __nodebug__))
+_mm_stream_pd(double *__p, __m128d __a)
+{
+  // Emscripten/SIMD.js does not have cache hinting.
+  _mm_store_pd(__p, __a);
+}
+
+static __inline__ void __attribute__((__always_inline__, __nodebug__))
+_mm_stream_si128(__m128i *__p, __m128i __a)
+{
+  // Emscripten/SIMD.js does not have cache hinting.
+  _mm_store_si128(__p, __a);
+}
+
+static __inline__ void __attribute__((__always_inline__, __nodebug__))
+_mm_stream_si32(int *__p, int __a)
+{
+  // No cache hinting available.
+  /* TODO: Add a build flag EMSCRIPTEN_SIMD_REQUIRE_ELEMENT_ALIGNMENT or something similar to avoid this.
+    Then could just do *__p = __a; */
+  struct __unaligned {
+    int __v;
+  } __attribute__((__packed__, __may_alias__));
+  ((struct __unaligned *)__p)->__v = __a;
+}
+
+typedef long long __attribute__((aligned(1))) align1_int64;
+static __inline__ void __attribute__((__always_inline__, __nodebug__))
+_mm_stream_si64(long long *__p, long long __a)
+{
+  // No cache hinting available.
+  emscripten_align1_int *__q = (emscripten_align1_int*)__p;
+  memcpy(__q, &__a, sizeof(long long));
```

Comment:
Same here; I imagine this would be simpler as a single store to a pointer with an alignment attribute.